frontend/packages/ux-editor/src/components/Properties/EditSubformTableColumns/ColumnElement/EditColumnElement/EditColumnElement.tsx

```diff
@@ -112,27 +114,36 @@ const EditColumnElementHeader = ({ columnNumber }: EditColumnElementHeaderProps)
   );
 };
 
-type EditColumnElementComponentSelectProps = {
+export type EditColumnElementComponentSelectProps = {
   components: FormItem[];
   onSelectComponent: (values: string[]) => void;
 };
-const EditColumnElementComponentSelect = ({
+export const EditColumnElementComponentSelect = ({
   components,
   onSelectComponent,
 }: EditColumnElementComponentSelectProps) => {
   const { t } = useTranslation();
   return (
     <StudioCombobox
       label={t('ux_editor.properties_panel.subform_table_columns.choose_component')}
+      description={t(
+        'ux_editor.properties_panel.subform_table_columns.choose_component_description',
+      )}
       size='sm'
       onValueChange={onSelectComponent}
       id='columncomponentselect'
     >
-      {components.map((comp: FormItem) => (
-        <StudioCombobox.Option key={comp.id} value={comp.id} description={comp.type}>
-          {comp.id}
+      {components.length > 0 ? (
+        components.map((comp: FormItem) => (
+          <StudioCombobox.Option key={comp.id} value={comp.id} description={comp.type}>
+            {comp.id}
+          </StudioCombobox.Option>
+        ))
+      ) : (
+        <StudioCombobox.Option value='' key={'noComponentsWithLabel'} disabled>
+          {t('ux_editor.properties_panel.subform_table_columns.empty_columns_message')}
         </StudioCombobox.Option>
```

Comment:
What do you think about extracting this inline code outside of the jsx rendering?
For example:

```
const subformComponentOptions = (
  components?.map((comp: FormItem) => (
    <StudioCombobox.Option key={comp.id} value={comp.id} description={comp.type}>
      {comp.id}
    </StudioCombobox.Option>
  )) || (
    <StudioCombobox.Option value='' key='noComponentsWithLabel'>
      {t('ux_editor.properties_panel.subform_table_columns.empty_columns_message')}
    </StudioCombobox.Option>
  )
);

....
....

<StudioCombobox>
  {subformComponentOptions}
</StudioCombobox>
```

To keep it consistent with other solutions in Altinn Studio, the selector doesn’t need to be disabled in this case, as there is a message option instead. See a similar solution on the process page below:

https://github.com/user-attachments/assets/6c24362e-6c49-4b43-91a2-32c9f7f03299

Reply:
it did not work with || we still need to check against lenght , i removed disabled and used as this :
```
    const subformComponentOptions =
    components.length > 0 ? (
      components.map((comp: FormItem) => (
        <StudioCombobox.Option key={comp.id} value={comp.id} description={comp.type}>
          {comp.id}
        </StudioCombobox.Option>
      ))
    ) : (
      <StudioCombobox.Option value='' key={'noComponentsWithLabel'}>
        {t('ux_editor.properties_panel.subform_table_columns.no_components_available_message')}
      </StudioCombobox.Option>
    );

```